src/unstructured_client/_hooks/custom/clean_server_url_hook.py

```diff
@@ -7,31 +7,61 @@
 from unstructured_client.httpclient import HttpClient
 
 
-class CleanServerUrlSDKInitHook(SDKInitHook):
-    """Hook fixing common mistakes by users in defining `server_url` in the unstructured-client"""
+def clean_server_url(base_url: str) -> str:
+    """Fix url scheme and remove the '/general/v0/general' path."""
+
+    if not base_url:
+        return ""
+    # -- add a url scheme if not present (urllib.parse does not work reliably without it)
+    if "http" not in base_url:
```

Comment:
I would parse the URL first and then check if a url has a scheme by using `parsed_url.scheme`. Then we can prepend `http://` or `https://`

Reply:
That comment has been around for a while - I tried parsing without the scheme and it does look like we need to make sure `http://` is there first.

```
In [2]: urlparse("localhost:8000/general/v0/general")
Out[2]: ParseResult(scheme='localhost', netloc='', path='8000/general/v0/general', params='', query='', fragment='')
```

Reply:
Oh I see, good to know!